src/mono/mono/mini/aot-compiler.c

```diff
@@ -12407,22 +12408,24 @@ emit_file_info (MonoAotCompile *acfg)
 
 	if (acfg->aot_opts.static_link) {
 		char symbol [MAX_SYMBOL_SIZE];
-		char *p;
 
 		/*
 		 * Emit a global symbol which can be passed by an embedding app to
 		 * mono_aot_register_module (). The symbol points to a pointer to the file info
 		 * structure.
 		 */
 		sprintf (symbol, "%smono_aot_module_%s_info", acfg->user_symbol_prefix, acfg->image->assembly->aname.name);
-
+#ifdef TARGET_WASM
+		acfg->static_linking_symbol = g_strdup (mono_fixup_symbol_name(symbol));
+#else
 		/* Get rid of characters which cannot occur in symbols */
```

Comment:
I wonder why this needs to be wasm-specific, shouldn't we do the same logic everywhere?

Reply:
It broke `iOS/tvOS` tests that's why I applied the logic only for wasm

Reply:
hmm that sounds concerning, I don't see why it should break iOS/tvOS

Reply:
This is what error looked like (from https://dev.azure.com/dnceng-public/public/_build/results?buildId=652242&view=logs&j=88e1dda0-08c6-5d85-6ee4-8e42aeff31b6&t=b4bb5080-7a46-5114-c100-ade75b904ada) 
<img width="1897" alt="ios_error" src="https://github.com/dotnet/runtime/assets/96171496/53ce4268-5861-42b4-bbb8-6d17ae442ce9">

My guess is that for iOS/tvOS in other places it still expects the previous logic. And I'm not sure that for other platforms there is a need to apply the change.

Reply:
then we need to find these places, the logic should be consistent